packages/core/useStorage/index.ts

```diff
@@ -218,7 +218,15 @@ export function useStorage<T extends(string | number | boolean | object | null)>
   }
 
   function update(event?: StorageEvent) {
-    if (event && event.key !== key)
+    if (event?.storageArea !== storage)
+      return
+
+    if (event?.key === null) {
```

Comment:
According to documentation, when `key` is `null`, it means that storage has been cleared